### PR TITLE
terraform plugin: relative source + direct main spec

### PR DESCRIPTION
**Pull request by mandelsoft**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement user
The terraform plugin now supports the generation of the main module directly in the deployment.yaml.
Additionally the source field may now be a path relative to the component folder.
```

### CODE REVIEW

**Comment by mandelsoft** [comment]
@Diaphteiros Could please have a look. It's a long time ago that I touched this project. May be you have some concerns.

File: plugins/terraform/plugin [hunk 1]
```diff
@@ -20,9 +20,43 @@ source "$SOWLIB/pluginutils"
...
+      fi
+    fi
+  fi
+else
+   if ! isAbsolutePath "$source"; then
+     if [ -d "$src/$source" ]; then
+       source="$CURRENT/$source"
+     else
+       source="$CURRENT/$source"
+     fi
```
Comment (Diaphteiros) [inline plugins/terraform/plugin:53]: This looks like either an unnecessary `if` or a mistake ...

**Review by Diaphteiros — CHANGES_REQUESTED** [review]
One small remark, looks good otherwise.

File: plugins/terraform/plugin (hunk 1 above)
Comment (mandelsoft) [reply plugins/terraform/plugin]: yupp, the first one must be $src/$source

**Review by Diaphteiros — APPROVED** [review]
/lgtm

**Pull request merged**